src/test/java/software/amazon/smithy/lsp/ext/CompletionsTest.java

```diff
@@ -106,6 +106,51 @@ public void multiFile() throws Exception {
         }
     }
 
+    @Test
+    public void multiFileV2() throws Exception {
+        Path baseDir = Paths.get(Completions.class.getResource("models/v2").toURI());
+        Path traitDefModel = baseDir.resolve("trait-def.smithy");
+        String traitDef = IoUtils.readUtf8File(traitDefModel);
+
+        Map<String, String> files = MapUtils.ofEntries(
+                MapUtils.entry("def1.smithy", "namespace test\nstring MyId"),
+                MapUtils.entry("bar/def2.smithy", "namespace test\nstructure Hello{}\ninteger MyId2"),
+                MapUtils.entry("foo/hello/def3.smithy", "namespace test\n@test()\n@trait\nstructure Foo {}"),
+                MapUtils.entry("foo/hello/def4.smithy", "namespace test\n@http()\noperation Bar{}"),
+                MapUtils.entry("trait-def.smithy", traitDef)
+        );
+
+        try (Harness hs = Harness.create(SmithyBuildExtensions.builder().build(), files)) {
+            SmithyProject proj = hs.getProject();
+            // Complete match
+            assertEquals(SetUtils.of("Foo"), completeNames(proj, "Foo", false));
+            // Partial match
+            assertEquals(SetUtils.of("MyId", "MyId2"), completeNames(proj, "MyI", false));
+            // Partial match (case insensitive)
+            assertEquals(SetUtils.of("MyId", "MyId2"), completeNames(proj, "myi", false));
+
+            // no matches
+            assertEquals(SetUtils.of(), completeNames(proj, "basdasdasdasd", false));
+            // empty token
+            assertEquals(SetUtils.of(), completeNames(proj, "", false));
+            // built-in
+            assertEquals(SetUtils.of("string", "String"), completeNames(proj, "Strin", false));
+            assertEquals(SetUtils.of("integer", "Integer"), completeNames(proj, "integer", false));
```

Comment:
Should this test be something like? If it is, the corresponding v1 test should also be changed.

```suggestion
            assertEquals(SetUtils.of("integer", "Integer"), completeNames(proj, "intege", false));
```

Reply:
Fixed in [0ed9978](https://github.com/awslabs/smithy-language-server/pull/61/commits/0ed9978ed9b3d58df77fb1af80310514f8d352bd).